.github/workflows/sync.yml

```diff
@@ -0,0 +1,54 @@
+name: "Check changes and sync"
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 5 * * *
+
+jobs:
+  check-and-sync:
+    runs-on: ubuntu-20.04
+    outputs:
+      all: ${{ steps.changes.outputs.all}}
+      c: ${{ steps.changes.outputs.c }}
+    steps:
+      - name: checkout tree-sitter-scala
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 10
+
+      - name: Generate parser from scratch
+        run: |
+          npm install
+          npm run build
+
+      - name: Check for changes
+        uses: tj-actions/verify-changed-files@v13
+        id: verify-changed-files
+        with:
+          files: |
+            src/grammar.json
+            src/node-types.json
+            src/parser.c
+            src/tree_sitter/parser.h
+
+      - name: Commit changes if necessary
+        if: steps.check-for-changes.outputs.files_changed == 'true'
+        run: |
+          git config user.name "GitHub"
```

Comment:
So github just provides an account you can use to do things like these?
Wild if so - I'm tempted to just merge this and run it manually to see if it works!

Reply:
Seems so, I took a lot of this from the way nvim-treesitter does this for their lock file.